server/api/shiftView.js

```diff
@@ -0,0 +1,178 @@
+      return 0
+    })
+
+    const formattedTimeBlocks = []
+
+    for (let i = 0; i < timeBlocks.length; i++) {
+      const timeBlockInfo = {
+        timeID: timeBlocks[i].timeId,
+        startTime: timeBlocks[i].timeStart,
+        endTime: timeBlocks[i].timeEnd,
+        positionName: timeBlocks[i].positionName,
+        assignedToID: timeBlocks[i].userId,
+        assignedToName: timeBlocks[i].userName
+      }
+
+      formattedTimeBlocks.push(timeBlockInfo)
+    }
+
+    res.status(200).json({
+      timeblocks: formattedTimeBlocks,
+      employees: formattedEmployees
+    }).end()
+  } catch (err) {
+    switch (err.message) {
+      // Data is either missing or invalid
+      case 'Missing/Invalid':
+        res.status(400)
+          .json(err.cause)
+        break
+      // User doe not exist in database
+      case 'Unauthorized':
+        res.status(401)
+          .json(err.cause)
+        break
+      default:
+        console.error(err)
+        // Some kind of error occurred, though not unique constraints
+        res.status(500)
+          .json({
+            error: {
+              status: 500,
+              message: 'Internal Server Error'
+            }
+          })
+          .end()
+    }
+  }
+}
+
+const shiftViewPOSTCallback = async (req, res) => {
+  try {
+    await verifyJWTAuthToken(req, res)
+
+    const shiftData = req.body
+
+    // get timeblock id and user id
+    if (shiftData.records?.length === 0) {
+      throw new Error('Missing/Invalid', { cause: { error: { status: 400, message: 'Missing Required Data' } } })
+    }
+
+    if (!shiftData?.isManager) {
+      throw new Error('Unauthorized', { cause: { error: { status: 401, message: 'Invalid Authorization Level' } } })
+    }
+
+    for (let i = 0; i < shiftData.records.length; ++i) {
+      if (!shiftData.records[i].timeBlockId || !shiftData.records[i].employeeId) {
+        console.log('error')
+        throw new Error('Missing/Invalid', { cause: { error: { status: 400, message: 'Missing Required Data' } } })
+      }
+
+      await assignShiftToUser(shiftData.records[i].timeBlockId, shiftData.records[i].employeeId)
+    }
+
+    // return success
+    res.status(200)
+      .json({
+        success: {
+          status: 200,
+          message: 'Assigned Successfully'
+        }
+      })
+      .end()
+  } catch (err) {
+    switch (err.message) {
+      // Data is either missing or invalid
+      case 'Missing/Invalid':
+        res.status(400)
+          .json(err.cause)
+        break
+      // User doe not exist in database
+      case 'Unauthorized':
+        res.status(401)
+          .json(err.cause)
+        break
+      default:
+        console.error(err)
+        // Some kind of error occurred, though not unique constraints
+        res.status(500)
+          .json({
+            error: {
+              status: 500,
+              message: 'Internal Server Error'
+            }
+          })
+          .end()
```

Comment:
Thoughts on making this block into a method in datahelper?
Example method in timeBlockData.js